synapse/storage/databases/state/store.py

```diff
@@ -418,10 +418,38 @@ async def store_state_group(
             The state group ID
         """
 
-        def _store_state_group_txn(txn: LoggingTransaction) -> int:
-            if current_state_ids is None:
-                # AFAIK, this can never happen
-                raise Exception("current_state_ids cannot be None")
+        if prev_group is None and current_state_ids is None:
+            raise Exception("current_state_ids and prev_group can't both be None")
+
+        if prev_group is not None and delta_ids is None:
+            raise Exception("delta_ids is None when prev_group is not None")
+
+        def insert_delta_group_txn(
+            txn: LoggingTransaction, prev_group: int, delta_ids: StateMap[str]
+        ) -> Optional[int]:
+            """If we have a delta we try and persist the new group as a delta.
```

Comment:
This function isn't conditional on whether we have a delta, so this is misleading.

```suggestion
            """Try and persist the new group as a delta.
            
            Requires that we have the state as a delta from a previous state group.
```